tests/foreman/maintain/test_health.py

```diff
@@ -0,0 +1,882 @@
+
+    :expectedresults: Check corrupted-roles should work and should handle corrupted roles
+                      by splitting filters as per updated values.
+
+    :CaseImportance: Medium
+
+    :BZ: 1703041, 1908846
+    """
+    # Check the filter created to verify the role, resource type, and permissions assigned.
+    role_name = 'test_role'
+    resource_type = gen_string("alpha")
+    sat_maintain.execute(f'hammer role create --name {role_name}')
+    sat_maintain.execute(
+        f'hammer filter create --role {role_name} --permissions view_hosts,console_hosts'
+    )
+    permission_name = r"'\''console_hosts'\''"
+    resource_type = rf"'\''{resource_type}'\''"
+    setup = sat_maintain.execute(
+        f'''sudo su - postgres -c "psql -d foreman -c 'UPDATE permissions SET
+            resource_type = {resource_type} WHERE name = {permission_name};'"'''
+    )
+    assert setup.status == 0
+    result = sat_maintain.execute(
+        r"hammer --output yaml filter list --search role=test_role | grep '\- Id:'"
+    )
+    assert 'Id' in result.stdout
+    # Run corrupted-roles check.
+    result = sat_maintain.cli.Health.check({'label': 'corrupted-roles', 'assumeyes': True})
+    assert result.status == 0
+    assert 'FAIL' in result.stdout
+    # Verify corrupted roles are fixed and new filter is created for updated resource_type.
+    result = sat_maintain.execute(
+        r"hammer --output yaml filter list --search role=test_role | grep '\- Id:'"
+    )
+    assert result.status == 0
+
+    @request.addfinalizer
+    def _finalize():
+        resource_type = r"'\''Host'\''"
+        assert (
+            sat_maintain.execute(
+                f'''sudo su - postgres -c "psql -d foreman -c 'UPDATE permissions SET
+                    resource_type = {resource_type} WHERE name = {permission_name};'"'''
+            ).status
+            == 0
+        )
+        assert sat_maintain.execute(f'hammer role delete --name {role_name}').status == 0
+
+
+@pytest.mark.include_capsule
+def test_positive_check_non_rh_packages(sat_maintain, request):
+    """Verify check-non-redhat-repository.
+
+    :id: 2b7a25bb-902c-4a87-bb71-f460dcce655c
+
+
+        1. satellite-maintain should be installed.
+        2. Configure custom repository and install custom package
+
+    :steps:
+        1. Run satellite-maintain health check --label non-rh-packages.
+        2. Verify presense of non-RH/custom package installed on system.
+
+    :BZ: 1869865
+
+    :expectedresults: check-non-rh-packages should work.
+
+    :CaseImportance: High
+    """
+    sat_maintain.create_custom_repos(custom_repo=settings.repos.yum_0.url)
+    assert (
+        sat_maintain.cli.Packages.install(packages='walrus', options={'assumeyes': True}).status
+        == 0
+    )
+    result = sat_maintain.cli.Health.check({'label': 'non-rh-packages'})
+    assert 'Found 1 unexpected non Red Hat Package(s) installed!' in result.stdout
```

Comment:
```
    Found 2 unexpected non Red Hat Package(s) installed!    
    Package : Vendor
    walrus-5.21-1.noarch : (none)
    hotfix-package-1.0.0-1.HOTFIX.noarch : (none)
```
This assertion is also failing due to the sed failure mentioned above (since the subsequent dnf command is not running).

Reply:
Updated the test.